src/main/java/no/digipost/monitoring/db/SlaveDbStatusCollector.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (C) Posten Norge AS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package no.digipost.monitoring.db;
+
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+/**
+ * This collector is able to say if the database is in recovery and thus queries master
+ */
+public class SlaveDbStatusCollector implements DbStatusCollector {
+
+    private static final String VALIDATION_QUERY = "SELECT pg_is_in_recovery()";
```

Comment:
This validation query will only work with postgreSQL. Fine by me, but maybe worth mentioning in javadoc.

Reply:
I will name it differently